build.gradle.kts

```diff
@@ -0,0 +1,148 @@
+    implementation("com.moandjiezana.toml:toml4j:0.7.2")
+    implementation("com.google.code.gson:gson:2.10.1")
+    implementation("net.i2p.crypto:eddsa:0.3.0")
+    implementation("commons-codec:commons-codec:1.16.0")
+
+    testImplementation("org.mockito:mockito-core:5.5.0")
+    testImplementation("com.squareup.okhttp3:mockwebserver:${okhttpVersion}")
+    testImplementation("junit:junit:4.13.2")
+    testImplementation("org.junit.vintage:junit-vintage-engine:5.10.0")
+}
+
+tasks {
+    test {
+        useJUnitPlatform()
+    }
+
+    jar {
+        manifest {
+            attributes["Implementation-Title"] = "stellar-sdk"
+            attributes["Implementation-Version"] = version
+        }
+        archiveFileName = "stellar-sdk.jar"
+    }
+
+    val uberJar by creating(Jar::class) {
+        // https://docs.gradle.org/current/userguide/working_with_files.html#sec:creating_uber_jar_example
+        manifest {
+            attributes["Implementation-Title"] = "stellar-sdk"
+            attributes["Implementation-Version"] = version
+        }
+        archiveClassifier = "uber"
+        archiveFileName = "stellar-sdk-uber.jar"
+        duplicatesStrategy = DuplicatesStrategy.EXCLUDE
+        from(sourceSets.main.get().output)
+        dependsOn(configurations.runtimeClasspath)
+        from({
+            configurations.runtimeClasspath.get().filter { it.name.endsWith("jar") }.map { zipTree(it) }
+        })
+    }
+
+    val javadocJar by creating(Jar::class) {
+        manifest {
+            attributes["Implementation-Title"] = "stellar-sdk"
+            attributes["Implementation-Version"] = version
+        }
+        archiveClassifier = "javadoc"
+        archiveFileName = "stellar-sdk-javadoc.jar"
+        dependsOn(javadoc)
+        from(javadoc.get().destinationDir)
+    }
+
+    javadoc {
+        destinationDir = file("javadoc")
+        isFailOnError = false
+        exclude("org/stellar/sdk/xdr/**")
+        options {
+            // https://docs.gradle.org/current/javadoc/org/gradle/external/javadoc/StandardJavadocDocletOptions.html
+            this as StandardJavadocDocletOptions
+            isSplitIndex = true
+            memberLevel = JavadocMemberLevel.PUBLIC
+            encoding = "UTF-8"
+        }
+    }
+
+    register<Copy>("updateGitHook") {
+        from("scripts/pre-commit.sh") { rename { it.removeSuffix(".sh") } }
+        into(".git/hooks")
+        doLast {
+            file(".git/hooks/pre-commit").setExecutable(true)
+        }
+    }
+
+    compileJava {
+        options.encoding = "UTF-8"
+    }
+
+    compileTestJava {
+        options.encoding = "UTF-8"
+    }
+}
+
+artifacts {
+    archives(tasks.jar)
+    archives(tasks["uberJar"])
+    archives(tasks["javadocJar"])
+}
+
+publishing {
+    publications {
+        create<MavenPublication>("mavenJava") {
+            from(components["java"])
+            artifact(tasks["uberJar"])
```

Comment:
what will the artifact `classifier` be for the uber jar? Assuming the non-uber jar will be the artifact with `classifier=jar` correct?

Reply:
Their classifiers are `''`, `'uber'`, and `'javadoc'`.

Reply:
Users can add them in three different ways.
```
    implementation("com.github.stellar:java-stellar-sdk:0.41.0")
    implementation("com.github.stellar:java-stellar-sdk:0.41.0:uber")
    implementation("com.github.stellar:java-stellar-sdk:0.41.0:javadoc")
```

Reply:
awesome, would you mind just adding a note for this in pending section in the CHANGELOG.md, so users can choose between the legacy behavior with dependency of `com.github.stellar:java-stellar-sdk:0.41.0:uber` or if they want 'thin' jar and transitive dependency management then `com.github.stellar:java-stellar-sdk:0.41.0`

Reply:
I have updated the changelog, but ubar jar is different from the previous shadow jar in that it does not relocate third-party dependencies.